Anastasia_PCS/Homework #1.js

```diff
@@ -0,0 +1,252 @@
+r : horizontal position of right edge;
+t : vertical position of top edge;
+b : vertical position of bottom edge.
+
+//a) Write an expression for the rectangle's area.
+//var l = 5;
+//var r = 5;
+//var t = 3; 
+//var b = 3;
+
+var area = (t-b)*(l-r);
+	//console.log(area); = 15
+
+
+//b) Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+if ((l-r)>(t-b)) {
+	console.log (true);
+}else{
+	console.log (false);
+};
+
+
+//c) Write an expression for the circumference of the biggest circle which can fit inside the rectangle. 
+//(Hint: you'll need logic similar to that in b.)
+
+if ((t-b) > (r-l)) {
+	var diameter = (r-1);
+} else {
+	diameter = (t-b);
+}
+var circumference = (diameter * Math.PI);
+console.log(circumference); 
+
+//or
+
+var isTall = (t-b) > (r-l);
+var isWide = (t-b) < (r-l);
+if (isTall) {
+	var diameter = (r-1);
+}
+else(isWide);{
+	diameter = (t-b); 
+}
+var circumference = (diameter * Math.PI);
+console.log(circumference); 
+
+//d) Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) 
+//the rectangle.
+
+var t = 7;
+var b = 3;
+var l = 2;
+var r = 8;
+
+var diameterSquared;
+var diameter;
+
+var topBottom = (t-b);
+var leftRight = (r-l);
+
+diameterSquared = Math.pow(leftRight, 2) + Math.pow(topBottom, 2);
+diameter = Math.sqrt(diameterSquared);
+console.log(diameter);
+
+//e) Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 
+//smaller rectangles, identical in shape but varying by position. Define four new variables describing the 
+//centermost small rectangle. (Hint: one of the many solutions is very similar to the solution of 2b above.)
+
+var t = 700;
+var b = 300;
+var l = 200; 
+var r = 800;
+
+var horizontal =  ((r-1)/3);
+var vertical = ((t-b)/3);
+
+ var newLeft = (1 + horizontal);
+ var newRight = (1 - horizontal);
+ var newBottom = (1 + vertical);
+ var newTop = (1 - vertical);
+
+console.log(newLeft, newRight, newBottom, newTop);
+
+//4) (25%)
+
+//Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+//Each square has integer coordinates (R,C) describing its row and column. Both values should be in the range 
+//0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+//Each square has a unique integer number N from 0 to 63. These numbers run sequentially left-to-right 
+//one row at a time, top to bottom. Therefore the upper-left square has N===0 and the bottom-right has 
+//N===63.
+
+//a) Given a particular R and C, find the corresponding N. That is, write an expression for variable 
+//N containing variables R and C. 
+
+
+var N = (8 * R) + C;
+
+
+//EXAMPLES: 
+//7,7 = 63
+//6,6 = 54
+
+//NOTE 0 - 63 means there are 64 spaces, so instead of multiplying by 7, I multiply by 8.
+
+//b) Given N, find R. Write an expression for R which contains N.
+
+var R = (N - (N % 8)) /8;
+
```

Comment:
This is really close! Looks like Dan wanted you to use the function math.floor (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor), which makes it much less complex. Essentially what this function does is round the solution down to the nearest whole number.  

The correct answer to this expression would be: R = math.floor(n/8). So, for example, if we use the number 9 in this expression:  var R = math.floor(9/8), math.floor would round the solution to 9/8 (which is 1.125) down to 1, which would give us the correct row, because the number 9 is in row 1 :)

Your answer is pretty much correct, it just has decimals at the end of some of the solutions for R, which aren't necessary.